src/http2_stream.erl

```diff
@@ -574,10 +580,17 @@ check_content_length(Stream) ->
         undefined ->
             ok;
         _Other ->
-            case Stream#stream_state.request_body_size =:= binary_to_integer(ContentLength) of
-                true ->
-                    ok;
-                false ->
+            try binary_to_integer(ContentLength) of
+                Integer ->
+                    case Stream#stream_state.request_body_size =:= Integer of
+                        true ->
+                            ok;
+                        false ->
+                            rst_stream_(?PROTOCOL_ERROR, Stream),
+                            rst_stream
+                    end
+            catch
+                _:_ ->
```

Comment:
But don't I want to send a rst_stream no matter why there's an error converting content-length to an integer?